### PR TITLE
tikv-ctl: pass db options to ldb

### PR DESCRIPTION
**Pull request by huachaohuang**
## What have you changed? (mandatory)

There are two problems in tikv-ctl related to the ldb command:
1. it relies on the position of the command line argument, which makes it impossible to parse other arguments outside of the ldb command
2. it doesn't pass the db options to ldb, which is required to support encryption environment

This PR solves these problems by adding "ldb" as a normal command line argument and passing the db options down to the ldb command.

## What are the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Manual

## Does this PR affect documentation (docs) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Refer to a related PR or issue link (optional)

https://github.com/tikv/tikv/pull/3936

### CODE REVIEW

File: docs/tools/tikv-control.md [hunk 1]
```diff
@@ -253,28 +253,28 @@ success!
...
+The ldb command line tool offers multiple data access and database administration commands. Some examples are listed below.
+For more information, refer to the help message displayed when running `tikv-ctl --ldb help` or check the documents from RocksDB.
 
 Examples of data access sequence:
 
 To dump an existing RocksDB in HEX:
 
 ```bash
-$ tikv-ctl ldb --db=/tmp/test_db dump --hex > /tmp/dbdump
+$ tikv-ctl --ldb='--hex --db=/tmp/db dump'
```
Comment (siddontang) [inline docs/tools/tikv-control.md]: em, IMO, I like the origin way - treating all the following args as the ldb args.

Comment (huachaohuang) [reply docs/tools/tikv-control.md]: Yes, I like that way too, but it makes us impossible to parse other arguments besides ldb, which is necessary to support encryption later.

Comment (Connor1996) [reply docs/tools/tikv-control.md]: How about treating all the args following `--` as the ldb args
like `tikv-ctl ldb xxx -- --hex --db=/tmp/db dump`

Comment (huachaohuang) [reply docs/tools/tikv-control.md]: But I don't think it is better, it implies that all the arguments behind `--` must be the arguments for ldb.

Comment (siddontang) [reply docs/tools/tikv-control.md]: seem it can't work if the db path has spaces.

Comment (siddontang) [reply docs/tools/tikv-control.md]: why not providing another tool like tikv-ldb?

Comment (Connor1996) [reply docs/tools/tikv-control.md]: I think this way is with common sense. Such like `cargo`, it uses`cargo test -- --xxx` to pass `--xxx` to test library.

Comment (huachaohuang) [reply docs/tools/tikv-control.md]: Done, I figure out a better solution to allow "ldb" as an external command with `clap`.

**Comment by huachaohuang** [comment]
@DorianZheng @Connor1996 @siddontang @zhangjinpeng1987 PTAL

File: docs/tools/tikv-control.md [hunk 2]
```diff
@@ -253,28 +252,28 @@ success!
...
+The ldb command line tool offers multiple data access and database administration commands. Some examples are listed below.
+For more information, refer to the help message displayed when running `tikv-ctl ldb` or check the documents from RocksDB.
 
 Examples of data access sequence:
 
 To dump an existing RocksDB in HEX:
 
 ```bash
-$ tikv-ctl ldb --db=/tmp/test_db dump --hex > /tmp/dbdump
+$ tikv-ctl ldb --hex --db=/tmp/db dump
```
Comment (siddontang) [inline docs/tools/tikv-control.md]: can we still use `> /tmp/dbdump` to redirect the output?

Comment (huachaohuang) [reply docs/tools/tikv-control.md]: Yes, of course, it's just a common output redirection, not specific to `ldb`.

**Pull request merged**